x/stakeibc/types/message_register_community_pool_rebate_test.go

```diff
@@ -0,0 +1,164 @@
+package types_test
+
+import (
+	"testing"
+
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/Stride-Labs/stride/v19/app/apptesting"
+	"github.com/Stride-Labs/stride/v19/x/stakeibc/types"
+)
+
+func TestMsgRegisterCommunityPoolRebate(t *testing.T) {
+	apptesting.SetupConfig()
+
+	validNotAdminAddress, invalidAddress := apptesting.GenerateTestAddrs()
+	validAdminAddress, ok := apptesting.GetAdminAddress()
+	require.True(t, ok)
+
+	validChainId := "chain-0"
+	validRebatePercentage := sdk.MustNewDecFromStr("0.1")
+	validLiquidStakeAmount := sdk.NewInt(1000)
+
+	tests := []struct {
+		name string
+		msg  types.MsgRegisterCommunityPoolRebate
+		err  string
+	}{
+		{
+			name: "valid message",
+			msg: types.MsgRegisterCommunityPoolRebate{
+				Creator:           validAdminAddress,
+				ChainId:           validChainId,
+				RebatePercentage:  validRebatePercentage,
+				LiquidStakeAmount: validLiquidStakeAmount,
+			},
+		},
+		{
+			name: "invalid address",
+			msg: types.MsgRegisterCommunityPoolRebate{
+				Creator:           invalidAddress,
+				ChainId:           validChainId,
+				RebatePercentage:  validRebatePercentage,
+				LiquidStakeAmount: validLiquidStakeAmount,
+			},
+			err: "invalid creator address",
+		},
+		{
+			name: "not admin address",
+			msg: types.MsgRegisterCommunityPoolRebate{
+				Creator:           validNotAdminAddress,
+				ChainId:           validChainId,
+				RebatePercentage:  validRebatePercentage,
+				LiquidStakeAmount: validLiquidStakeAmount,
+			},
+			err: "not an admin",
+		},
+		{
+			name: "invalid chain ID",
+			msg: types.MsgRegisterCommunityPoolRebate{
+				Creator:           validAdminAddress,
+				ChainId:           "",
+				RebatePercentage:  validRebatePercentage,
+				LiquidStakeAmount: validLiquidStakeAmount,
+			},
+			err: "chain ID must be specified",
+		},
+		{
+			name: "invalid rebate percentage - nil",
+			msg: types.MsgRegisterCommunityPoolRebate{
+				Creator:           validAdminAddress,
+				ChainId:           validChainId,
+				LiquidStakeAmount: validLiquidStakeAmount,
+			},
+			err: "rebate percentage, must be between [0, 1)",
```

Comment:
if we accept new bounds, update to [0,1]

Reply:
done

Reply:
lgtm